File: licentainangular.client/src/app/models/Subcomanda.ts
import { Produs } from './Produs';
import { Comanda } from './Comanda'
// models/Subcomanda.ts
export interface Subcomanda {
  IdSubcomanda: number;
  IdProdus: number;
  Produs?: any;       // navigational property, opțional
  IdComanda: number;
  Comanda?: any;      // navigational property, opțional
  TotalSubproduse: number;  // => important: corespunde "TotalSubproduse" din C#
}

File: LicentaInAngular.Server/DataLayer/Models/Subcomanda.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace LicentaInAngular.Server.Models
{
    public class Subcomanda
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

        public int IdSubcomanda { get; set; }


        [ForeignKey("Produs")]
        public int IdProdus { get; set; }
        public Produs Produs { get; set; }

        [Required]
        public int TotalSubproduse { get; set; }


        [ForeignKey("Comanda")]
        public int IdComanda { get; set; }
        public Comanda Comanda { get; set; }
    }
}